articles/055_ros_parameter_design.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: default
+title: Parameter API design in ROS
+permalink: articles/ros_parameters.html
+abstract:
+  This article is proposed design for the interfaces for interacting with parameters in ROS 2.0.
+  We focus here on specifying the system design and leave the implementation unspecified.
+author: '[Tully Foote](https://github.com/tfoote)'
+published: true
+---
+
+* This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+In ROS 1 the parameters were implemented in a 'blackboard model' with unrestricted read and write access from all nodes.
+The data model proved useful in many cases, but there were many cases where the lack of control or ownership proved to be a problem.
+One of the common shortcomings was for setting parameters on drivers.
+A tool called dynamic_reconfigure was developed to service this use case.
+It provided a service based interface to interact with parameters of other nodes.
+
+### Other resources
+
+Other resources related to the parameter design process for ROS 2.0 include:
+
+- Gonzalo's research on parameters.
+  - Discussion: https://groups.google.com/forum/#!topic/ros-sig-ng-ros/YzCmoIsN0o8 and https://groups.google.com/forum/#!searchin/ros-sig-ng-ros/parameter/ros-sig-ng-ros/fwDBcei5Ths/L6ORPfjUDXYJ
+  - Protoype: https://github.com/abellagonzalo/dynamic_config
+  - Final Notes:  [http://wiki.ros.org/sig/NextGenerationROS/Parameters](http://wiki.ros.org/sig/NextGenerationROS/Parameters)
+- Thibault's nodeparam draft REP: https://github.com/tkruse/rep/blob/nodeparam/nodeparam-REP.rst
+
+## Ideal System
+
+It is useful to consider the ideal system to understand how it relates to the current system and how a new system could work.
+We'd like to support both the use cases of ROS 1.0 built in parameters as well as dynamic parameters.
+An ideal parameter system would have the qualities laid out in the following paragraphs.
+
+### Accept parameter values
```

Comment:
I'd call this section "set" and the next "get" parameter values. This makes it line up with "unset" and it avoids "accept" which implies that it could have not been accepted ("reject"), which is not covered in the following paragraph.